src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImplTest.java

```diff
@@ -83,64 +130,41 @@ public void setUp() {
-    final Map<String, Parameter> expectedParameters = ImmutableMap.of(
-            "welcome_message_text", new Parameter()
-                    .setDefaultValue(ParameterValue.of("welcome to app"))
-                    .setConditionalValues(ImmutableMap.<String, ParameterValue>of(
-                            "ios_en", ParameterValue.of("welcome to app en")
-                    ))
-                    .setDescription("text for welcome message!"),
-            "header_text", new Parameter()
-                    .setDefaultValue(ParameterValue.inAppDefault())
-    );
-    final Map<String, ParameterGroup> expectedParameterGroups = ImmutableMap.of(
-            "new menu", new ParameterGroup()
-                    .setDescription("New Menu")
-                    .setParameters(ImmutableMap.of(
-                            "pumpkin_spice_season", new Parameter()
-                                    .setDefaultValue(ParameterValue.of("true"))
-                                    .setDescription("Whether it's currently pumpkin spice season.")
-                            )
-                    )
-    );
-    final List<Condition> expectedConditions = ImmutableList.of(
-            new Condition("ios_en", "device.os == 'ios' && device.country in ['us', 'uk']")
-                    .setTagColor(TagColor.INDIGO),
-            new Condition("android_en",
-                    "device.os == 'android' && device.country in ['us', 'uk']")
-                    .setTagColor(TagColor.UNSPECIFIED)
-    );
-    final Version expectedVersion = new Version(new TemplateResponse.VersionResponse()
-            .setVersionNumber("17")
-            .setUpdateOrigin("ADMIN_SDK_NODE")
-            .setUpdateType("INCREMENTAL_UPDATE")
-            .setUpdateUser(new TemplateResponse.UserResponse()
-                    .setEmail("firebase-user@account.com")
-                    .setName("dev-admin")
-                    .setImageUrl("http://image.jpg"))
-            .setUpdateTime("2020-11-03T20:24:15.045123456Z")
-            .setDescription("promo config")
-    );
-
-    Template expectedTemplate = new Template()
-            .setETag(TEST_ETAG)
-            .setParameters(expectedParameters)
-            .setConditions(expectedConditions)
-            .setParameterGroups(expectedParameterGroups)
-            .setVersion(expectedVersion);
 
     assertEquals(TEST_ETAG, receivedTemplate.getETag());
-    assertEquals(expectedTemplate, receivedTemplate);
-    assertEquals(1604435055000L, receivedTemplate.getVersion().getUpdateTime());
+    assertEquals(EXPECTED_TEMPLATE, receivedTemplate);
+    assertEquals(1605423446000L, receivedTemplate.getVersion().getUpdateTime());
     checkGetRequestHeader(interceptor.getLastRequest());
   }
 
+  @Test
+  public void testGetTemplateWithTimestampUpToNanosecondPrecision() throws Exception {
+    for (int i = 1; i < 10; i++) {
+      String fractionalSecs = "342763941".substring(0, i);
```

Comment:
This is a bit redundant. Just test for a few timestamp formats.

```
String[] timestamps = new String[] {
  // 3 digits,
  // 6 digits,
  // 9 digits,
}
```

Reply:
Thanks! Updated!